src/quick-lint-js/fe/lex.cpp

```diff
@@ -432,8 +432,16 @@ bool Lexer::try_parse_current_token() {
     break;
 
   case '<':
-    if (this->input_[1] == '!' && this->input_[2] == '-' &&
-        this->input_[3] == '-') {
+  // checking for EJS or PHP syntax
+    if((this->input_[0] == '<' && (this->input_[1] == '%')) || (this->input_[0] == '?')){
```

Comment:
Nit: Checking `this->input_[0] == '<'` is redundant. (Line 434 already checks for '<'.)